.eslintrc.js

```diff
@@ -11,36 +11,42 @@ module.exports = {
     browser: true,
     node: true,
   },
-  extends: ['plugin:import/recommended', 'airbnb', 'prettier', 'prettier/react'],
-  parser: 'babel-eslint',
+  extends: [
+    'plugin:import/recommended',
+    'plugin:import/typescript',
+    'airbnb-typescript',
+    'prettier',
+    'prettier/react',
+    'prettier/@typescript-eslint',
+  ],
+  parser: '@typescript-eslint/parser',
   parserOptions: {
-    ecmaVersion: 7,
-    sourceType: 'module',
+    // project: './tsconfig.json',
   },
-  plugins: ['babel', 'material-ui', 'react-hooks'],
+  plugins: ['material-ui', 'react-hooks', '@typescript-eslint'],
   settings: {
     'import/resolver': {
       webpack: {
         config: path.join(__dirname, './docs/webpackBaseConfig.js'),
       },
     },
   },
-  /**
-   * Sorted alphanumerically within each group. built-in and each plugin form
-   * their own groups.
-   */
+  // Sorted alphanumerically
```

Comment:
Please leave this by groups to avoid diff noise. It makes it easier to go through the rules by documentation.

Reply:
I'm not sure to follow, the rules are still sorted by group.

Reply:
They're not. You have `'consistent-this'` (base rule), `jsx-a11y/*` (`jsx-a11`) and then `max-classes-per-file` (base rule again).

Reply:
Hum, that's true. I guess I was blind to it, I like how sorting asc prevents to have discussion around what's the best way to order these things. At least not base rules are grouped correctly.

Reply:
I like how sorting by group and alphanumeric prevents the discussion as well. What sorting we choose is arbitrary. The previous sorting made it easier to work with dependency upgrades. You're just changing it for the sake of it. This is ill fit for a PR that adds and removes so many rules since it's not obvious what rules were actually removed but just moved.

Reply:
Ok, I will work on reducing the diff

Reply:
I have restored the previous grouping logic